content/rust/concepts/generics/generics.md

```diff
@@ -0,0 +1,115 @@
+---
+Title: ‘Rust Generics'
+Description: 'Threads allow for flexible function and data type definitions.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Generics'
+  - 'Types'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+**Generics** in Rust allows developers to write flexible, reusable code that can work with many data types. Generics can be used to create functions, data structures, and methods that operate over a variety of types while still providing compile-time type safety.
+
+## Defining a Generic Function
+
+A generic function can be defined by specifying a type parameter inside angle brackets `<>`. This type parameter can then be used as a stand-in for any type. By convention, most rust developers use `T`, which is short for `Type`. The following example demonstrates how to define a generic function:
+
+```rust
+fn my_function<T>(value: T) -> T {
+    ...
+}
+```
+
+In the above example, `my_function` is a generic function that takes a single parameter `value` of any type `T`.
+
+The next example illustrates a generic function that returns a value:
+
+```rust
+fn test<T>(a: T) -> T {
+    return a;
+}
+
+fn main() {
+    println!("{}", test(8));
+}
+```
+
+The above example will output the following:
+
+```shell
+8
+```
+
+## Using a Generic Data Structure
+
+Similarly, you can define `struct` or `enum` with generics. The following example showcases defining a `struct` with generics.
+
+```rust
+struct Coordinates<T> {
+    x: T,
+    y: T,
+}
+
+fn main() {
+    let int_coord = Coordinates {x: 5, y: 4};
+    let float_coord = Coordinates {x: 5.01, y: 4.10};
+    println!("int_coord: ({}, {})", int_coord.x, int_coord.y);
+    println!("float_coord: ({}, {})", float_coord.x, float_coord.y);
+}
+```
+
+The above example will give the following output:
+
+```shell
+int_coord: (5, 4)
+float_coord: (5.01, 4.1)
+```
+
+> **Note:** For the above example, ensure that `x` and `y` are of the same type, else there will be a mismatched types error.
+
+You can also define `struct` or `enum` with multiple generic type parameters. The following example showcases how to do so.
```

Comment:
We dont use pronouns and first-person speech. So use of words like `you` `we` etc aren't allowed please refactor the sentence.